### ISSUE TITLE
Click on distant /c link crashes immediately

### ISSUE THREAD

**Issue by pguimier** [dessalines/jerboa#812]
**Jerboa Version**
0.0.34

**Describe the bug**
All links with the form /c/community@instance.here giving this [Some text here](/c/community@instance.here), when clicked, crash immediately the application. This type of link is used to open in local instance a community that is hosted in a distant federated instance.

**To Reproduce**

1. Go to https://lemmy.world/post/406302 on jerboa
2. Click on 'Books' link (third link)
3. crash

I can't provide logcat logs now

**Comment by ReddishPython** [comment]
Happens in 0.0.33 from Google play store too

**Comment by MV-GH** [comment]
Still happens in main and in navigation enhancements x2

<details>
<summary>Stack trace</summary>

```

 Exception dispatching input event.
2023-06-22 15:00:25.419  2707-2707  MessageQueue-JNI        com.jerboa.debug                     E  Exception in MessageQueue callback: handleReceiveCallback
2023-06-22 15:00:25.423  2707-2707  MessageQueue-JNI        com.jerboa.debug                     E  java.lang.IllegalStateException: no event down from INITIALIZED in component androidx.navigation.NavBackStackEntry@3a392f20
                                                                                                    	at androidx.lifecycle.LifecycleRegistry.moveToState(LifecycleRegistry.kt:126)
                                                                                                    	at androidx.lifecycle.LifecycleRegistry.setCurrentState(LifecycleRegistry.kt:106)
                                                                                                    	at androidx.navigation.NavBackStackEntry.updateState(NavBackStackEntry.kt:183)
                                                                                                    	at androidx.navigation.NavBackStackEntry.setMaxLifecycle(NavBackStackEntry.kt:156)
                                                                                                    	at androidx.navigation.NavController.updateBackStackLifecycle$navigation_runtime_release(NavController.kt:987)
                                                                                                    	at androidx.navigation.NavController.dispatchOnDestinationChanged(NavController.kt:892)
                                                                                                    	at androidx.navigation.NavController.navigate(NavController.kt:1730)
                                                                                                    	at androidx.navigation.NavController.navigate(NavController.kt:1662)
                                                                                                    	at androidx.navigation.NavController.navigate(NavController.kt:1984)
                                                                                                    	at androidx.navigation.NavController.navigate$default(NavController.kt:1979)
                                                                                                    	at com.jerboa.UtilsKt.openLink(Utils.kt:381)
                                                                                                    	at com.jerboa.ui.components.common.MarkdownHelper$init$1.configureConfiguration$lambda$0(MarkdownHelper.kt:148)
                                                                                                    	at com.jerboa.ui.components.common.MarkdownHelper$init$1.$r8$lambda$Q5mfLV7gF7QKSMOOicLPF0X2nxM(Unknown Source:0)
                                                                                                    	at com.jerboa.ui.components.common.MarkdownHelper$init$1$$ExternalSyntheticLambda0.resolve(Unknown Source:6)
                                                                                                    	at io.noties.markwon.core.spans.LinkSpan.onClick(LinkSpan.java:30)
                                                                                                    	at android.text.method.LinkMovementMethod.onTouchEvent(LinkMovementMethod.java:232)
                                                                                                    	at io.noties.markwon.ext.tables.TableAwareMovementMethod.onTouchEvent(TableAwareMovementMethod.java:118)
                                                                                                    	at android.widget.TextView.onTouchEvent(TextView.java:10981)
                                                                                                    	at android.view.View.dispatchTouchEvent(View.java:14381)
                                                                                                    	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3151)
                                                                                                    	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2830)
                                                                                                    	at androidx.compose.ui.input.pointer.PointerInteropFilter_androidKt$pointerInteropFilter$3.invoke(PointerInteropFilter.android.kt:113)
                                                                                                    	at androidx.compose.ui.input.pointer.PointerInteropFilter_androidKt$pointerInteropFilter$3.invoke(PointerInteropFilter.android.kt:105)
                                                                                                    	at androidx.compose.ui.input.pointer.PointerInteropFilter$pointerInputFilter$1$dispatchToView$3.invoke(PointerInteropFilter.android.kt:309)
                                                                                                    	at androidx.compose.ui.input.pointer.PointerInteropFilter$pointerInputFilter$1$dispatchToView$3.invoke(PointerInteropFilter.android.kt:294)
                                                                                                    	at androidx.compose.ui.input.pointer.PointerInteropUtils_androidKt.toMotionEventScope-ubNVwUQ(PointerInteropUtils.android.kt:81)
                                                                                                    	at androidx.compose.ui.input.pointer.PointerInteropUtils_androidKt.toMotionEventScope-d-4ec7I(PointerInteropUtils.android.kt:35)
                                                                                                    	at androidx.compose.ui.input.pointer.PointerInteropFilter$pointerInputFilter$1.dispatchToView(PointerInteropFilter.android.kt:294)
                                                                                                    	at androidx.compose.ui.input.pointer.PointerInteropFilter$pointerInputFilter$1.onPointerEvent-H0pRuoY(PointerInteropFilter.android.kt:229)
                                                                                                    	at androidx.compose.ui.node.BackwardsCompatNode.onPointerEvent-H0pRuoY(BackwardsCompatNode.kt:374)
                                                                                                    	at androidx.compose.ui.input.pointer.Node.dispatchMainEventPass(HitPathTracker.kt:296)
                                                                                                    	at androidx.compose.ui.input.pointer.Node.dispatchMainEventPass(HitPathTracker.kt:301)
                                                                                                    	at androidx.compose.ui.input.pointer.Node.dispatchMainEventPass(HitPathTracker.kt:301)
                                                                                                    	at androidx.compose.ui.input.pointer.Node.dispatchMainEventPass(HitPathTracker.kt:301)
                                                                                                    	at androidx.compose.ui.input.pointer.Node.dispatchMainEventPass(HitPathTracker.kt:301)
                                                                                                    	at androidx.compose.ui.input.pointer.Node.dispatchMainEventPass(HitPathTracker.kt:301)
                                                                                                    	at androidx.compose.ui.input.pointer.Node.dispatchMainEventPass(HitPathTracker.kt:301)
                                                                                                    	at androidx.compose.ui.input.pointer.Node.dispatchMainEventPass(HitPathTracker.kt:301)
                                                                                                    	at androidx.compose.ui.input.pointer.NodeParent.dispatchMainEventPass(HitPathTracker.kt:183)
                                                                                                    	at androidx.compose.ui.input.pointer.HitPathTracker.dispatchChanges(HitPathTracker.kt:102)
2023-06-22 15:00:25.423  2707-2707  MessageQueue-JNI        com.jerboa.debug                     E  	at androidx.compose.ui.input.pointer.PointerInputEventProcessor.process-BIzXfog(PointerInputEventProcessor.kt:98)
                                                                                                    	at androidx.compose.ui.platform.AndroidComposeView.sendMotionEvent-8iAsVTc(AndroidComposeView.android.kt:1361)
                                                                                                    	at androidx.compose.ui.platform.AndroidComposeView.handleMotionEvent-8iAsVTc(AndroidComposeView.android.kt:1307)
                                                                                                    	at androidx.compose.ui.platform.AndroidComposeView.dispatchTouchEvent(AndroidComposeView.android.kt:1246)
                                                                                                    	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3151)
                                                                                                    	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2830)
                                                                                                    	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3151)
                                                                                                    	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2830)
                                                                                                    	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3151)
                                                                                                    	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2830)
                                                                                                    	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3151)
                                                                                                    	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2830)
                                                                                                    	at com.android.internal.policy.DecorView.superDispatchTouchEvent(DecorView.java:501)
                                                                                                    	at com.android.internal.policy.PhoneWindow.superDispatchTouchEvent(PhoneWindow.java:1875)
                                                                                                    	at android.app.Activity.dispatchTouchEvent(Activity.java:4225)
                                                                                                    	at com.android.internal.policy.DecorView.dispatchTouchEvent(DecorView.java:459)
                                                                                                    	at android.view.View.dispatchPointerEvent(View.java:14655)
                                                                                                    	at android.view.ViewRootImpl$ViewPostImeInputStage.processPointerEvent(ViewRootImpl.java:6278)
                                                                                                    	at android.view.ViewRootImpl$ViewPostImeInputStage.onProcess(ViewRootImpl.java:6065)
                                                                                                    	at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:5520)
                                                                                                    	at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:5577)
                                                                                                    	at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:5543)
                                                                                                    	at android.view.ViewRootImpl$AsyncInputStage.forward(ViewRootImpl.java:5695)
                                                                                                    	at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:5551)
                                                                                                    	at android.view.ViewRootImpl$AsyncInputStage.apply(ViewRootImpl.java:5752)
                                                                                                    	at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:5524)
                                                                                                    	at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:5577)
                                                                                                    	at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:5543)
                                                                                                    	at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:5551)
                                                                                                    	at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:5524)
                                                                                                    	at android.view.ViewRootImpl.deliverInputEvent(ViewRootImpl.java:8404)
                                                                                                    	at android.view.ViewRootImpl.doProcessInputEvents(ViewRootImpl.java:8355)
                                                                                                    	at android.view.ViewRootImpl.enqueueInputEvent(ViewRootImpl.java:8315)
                                                                                                    	at android.view.ViewRootImpl$WindowInputEventReceiver.onInputEvent(ViewRootImpl.java:8540)
                                                                                                    	at android.view.InputEventReceiver.dispatchInputEvent(InputEventReceiver.java:241)
                                                                                                    	at android.os.MessageQueue.nativePollOnce(Native Method)
                                                                                                    	at android.os.MessageQueue.next(MessageQueue.java:335)
                                                                                                    	at android.os.Looper.loop(Looper.java:193)
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:8061)
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method)
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:656)
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:967)

```

</details>

**Comment by beatgammit** [comment]
I'm having trouble testing this for a few reasons:

- I don't have an account on lemmy.world
- I don't know how to navigate to a specific post through the app and search doesn't work
- I apparently can't view source through the website if I'm not logged in

I can probably hack around it, and maybe this belongs as a feature request, I just figured I'd ask here since it's directly relevant to reproducing this issue.

**Comment by MV-GH** [comment]
I did post hte raw markdown see previous message, and you know whats weird now it actually worked

**Comment by MV-GH** [comment]
Solved with #1023
